core/ordersync/ordersync.go

```diff
@@ -445,46 +457,52 @@ func parseResponseWithSubprotocol(subprotocol Subprotocol, providerID peer.ID, r
 	}, nil
 }
 
-func (s *Service) getOrdersFromPeer(ctx context.Context, providerID peer.ID) error {
+func (s *Service) getOrdersFromPeer(ctx context.Context, providerID peer.ID, nextRequest *rawRequest) (*rawRequest, error) {
```

Comment:
Maybe in the scope of the `getOrdersFromPeer` function we should name this `firstRequest`? It represents the first request that `getOrdersFromPeer` will send to the provider. Subsequent requests are determined by the subprotocol (i.e. the return value of `HandleOrderSyncResponse`).

```suggestion
func (s *Service) getOrdersFromPeer(ctx context.Context, providerID peer.ID, firstRequest *rawRequest) (*rawRequest, error) {
```